src/components/presentational/Chat/Chat.tsx

```diff
@@ -0,0 +1,89 @@
+import parse from 'html-react-parser';
+
+import UnstyledButton from '../UnstyledButton';
+
+import './Chat.css';
+
+export type ChatMessageType = "sent" | "received";
+
+export interface ChatProps {
+    innerRef?: React.Ref<HTMLDivElement>;
+    messages: ChatMessage[];
+    onSendMessage: (newMessage: string) => void;
+    loading?: string;
+}
+
+export interface ChatMessage {
+    icon?: React.JSX.Element;
+    content: string;
+    type: ChatMessageType;
+}
+
+const md = new MarkdownIt();
+
+export default function (props: ChatProps) {
+
+    const [currentUserMessage, setCurrentUserMessage] = useState('');
+    const logRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (logRef.current) {
+            logRef.current.scrollTop = logRef.current.scrollHeight;
+        }
+    }, [props.messages]);
+
+    const sendMessage = async function () {
+        let newMessage = currentUserMessage;
+        props.onSendMessage(newMessage);
+        setCurrentUserMessage('');
+    }
+
+    return (
+        <div className="mdhui-chat" ref={props.innerRef}>
+            <div className="mdhui-chat-log">
+                <div className="mdhui-chat-messages" ref={logRef}>
+                    {props.messages.map((message, index) => {
+                        if (message.type === "sent") {
+                            return <div key={index} className="mdhui-chat-message mdhui-chat-sent-message-row">
+                                <div className="mdhui-chat-sent-message">
+                                    {parse(md.render(message.content))}
+                                </div>
+                            </div>
+                        }
+                        else {
+                            return <div key={index} className="mdhui-chat-message mdhui-chat-received-message-row">
+                                {message.icon}
+                                <div className="mdhui-chat-received-message">{parse(md.render(message.content))}</div>
+                            </div>
+                        }
+                    })}
+                    {props.loading && <div className="mdhui-chat-message-loading">
+                        <FontAwesomeSvgIcon icon={faSpinner} spin={true} />{props.loading}
+                    </div>}
+                </div>
+            </div>
+            <div className="mdhui-chat-input">
+                <div className="mdhui-chat-input-group">
+                    <input
+                        type="text"
+                        value={currentUserMessage}
+                        onChange={(e) => setCurrentUserMessage(e.target.value)}
+                        onKeyDown={(e) => {
+                            if (e.key === 'Enter') {
+                                sendMessage();
+                            }
+                        }}
+                    />
+                    <UnstyledButton onClick={sendMessage} className="mdhui-chat-send-button">
```

Comment:
Would disable the send and also make sure no event fires on enter when there's no message entered (blank string)